### DISCUSSION TITLE
How to implement in MLX the "expand" torch function ?

### DISCUSSION THREAD

**Discussion by thegodone** [ml-explore/mlx#1530] [Q&A]
I want to use the myarray.unsqueeze(-2).expand(a,b,c,d) I use this in mlx so far two steps myarray = mx.expand_dims(myarray, -2) and myarray = mxrepeat(myarray, c) for example than gave me same shape as torch. Is there a faster way or a better alternative ?

**Comment by angeloskath** [comment]
Actually when you `repeat` a singleton dimension there is no copy, it is a simple broadcast as above.
